app/js/HttpController.js

```diff
@@ -188,6 +188,17 @@ module.exports = HttpController = {
     })
   },
 
+  patchDoc(req, res, next) {
+    const { project_id, doc_id } = req.params
+    logger.log({ project_id, doc_id }, 'patching doc')
+    DocManager.patchDoc(project_id, doc_id, req.body, function (error) {
```

Comment:
It might be a bit paranoid, but I would avoid passing the body as is and, instead, explicitly extract the properties we're interested in: `name`, `deleted` and `deletedAt`. I know that Joi checks for unknown properties, but I had to look it up to be sure, and it's easy to imagine someone setting `allowUnknown` to true on Joi validators (one reason in another context would be to allow UTM params on query strings without having to explicitly allow them), so I wouldn't mind an extra layer of precaution here.

Reply:
That feels like duplicated effort on request validation. We should trust the outer level to do the right thing. Would you be fine with adding an acceptance test with an unknown field instead?

Reply:
I see that less as validation than as being explicit about the parameters that are sent to that method, particularly given that this object is going to be sent unmodified to Mongo. If you feel strongly about it, I'm ok leaving it as is because it's not broken at the moment, but I feel like this is fragile. An acceptance test would help, indeed, but it's probably less work to extract parameters from the body.

Reply:
Done in https://github.com/overleaf/docstore/pull/93/commits/480623ea871a7d2ca90a885de697ddd1bbf41f42